axelor-account/src/main/java/com/axelor/apps/account/service/umr/UmrNumberServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2018 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.axelor.apps.account.service.umr;
+
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+
+import com.axelor.apps.base.db.Partner;
+
+import net.fortuna.ical4j.model.Date;
+
+public class UmrNumberServiceImpl implements UmrNumberService {
+
+  @Override
+  public String getUmrNumber(Partner partner) {
+    return getUmrNumber(partner, false);
+  }
+
+  @Override
+  public String getUmrNumber(Partner partner, boolean isRecoveryContractLine) {
+
+    String rumNumber = "";
+
+    if(isRecoveryContractLine) {
+      rumNumber += "++";
+    }
+
+    rumNumber += partner.getId();
```

Comment:
Better to use partner.getPartnerSeq here

Reply:
done